new-website/utils/tutorials/export_tutorials.py

```diff
@@ -0,0 +1,182 @@
+import {{useEffect}} from "react";
+import scrollnav from "scrollnav";
+
+const {component_name} = () => {{
+
+useEffect(() => {{
+        document.getElementsByClassName('scroll-nav')[0]?.remove();
+        const content = document.querySelector(".notebook");
+        const insertTarget = document.querySelector(".notebook");
+
+        if (insertTarget && content) {{
+            scrollnav.init(content, {{
+                sections: "h1, h2", insertTarget: insertTarget, insertLocation: "after",
+            }});
+        }}
+
+        MathJax?.Hub?.Queue(["Typeset", MathJax.Hub]);
+    }}, []);
+
+return <div
+    className="overflow-x-scroll"
+    dangerouslySetInnerHTML={{{{__html: `${{innerHTML.html}} ${{notebookStyles}}`,}}}}
+></div>
+}}
+
+{component_name}.Layout = TutorialLayout;
+
+export default {component_name};
+"""
+
+    return generated_component
+
+
+def parse_and_export_tutorials():
+    """
+    Parses HTML files containing DeepChem tutorials and converts them into React components and JSON data that is used to
+    display tutorials on the website
+    """
+    tutorials = []
+
+    # Open 'notebooks.txt' file and iterate over each file name
+    with open('./notebooks.txt', 'r') as notebook_files:
+        for file in notebook_files:
+
+            # Open corresponding html file located in the 'html-notebooks' directory
+            with open(f'./html-notebooks/{file.strip()}', 'r', encoding="utf-8") as f:
+                tutorial = {}
+                soup = BeautifulSoup(f.read(), 'html.parser')
+                body = soup.body
+
+                # Get the title of the tutorial from the first h1 tag in the html file
+                title = soup.find('h1').text
+
+                # Remove the 'id' attribute from all the h1 and h2 tags in the html file
+                headings = body.find_all(['h1', 'h2'])
+                for heading in headings:
+                    del heading['id']
+
+                # Add the class 'overflow-x-scroll' to all the pre tags in the html file
+                code_cells = body.find_all('pre')
+                for code_cell in code_cells:
+                    code_cell['class'] = code_cell.get(
+                        'class', []) + ['overflow-x-scroll']
+
+                # Hide the body tag of the html file
+                soup.find('body').hidden = True
+
+                # Get the prettified html of the body tag
+                html = soup.body.prettify()
+
+                # Generate a url-friendly name for the tutorial and a valid component name for the javascript file
+                file_name = file.rsplit(".")[0]
+                urlified_file_name = to_valid_url_path(file_name)
+                component_name = to_valid_identifier(file_name)
+
+                # Write the tutorial html to a js file located in the 'data/tutorials' directory
+                with open(f'../../deepchem/data/tutorials/{urlified_file_name}.js', 'w', encoding="utf-8") as data_file:
```

Comment:
This makes an assumption about where the `deepchem/` repo is checked out? If so, it should be documented in the docstring and the README